pkg/cli/admin/nodeimage/create.go

```diff
@@ -345,6 +375,22 @@ func (o *CreateOptions) copyArtifactsFromNodeJoinerPod() error {
 	return rsyncOptions.RunRsync()
 }
 
+func (o *CreateOptions) isoName() string {
+	isoNameFormat := "node.%s.iso"
+	var architectureFileName string
+	switch o.CPUArchitecture {
+	case string(machineconfig.Arm64):
+		architectureFileName = string(machineconfig.Aarch64)
+	case string(machineconfig.Amd64):
+		architectureFileName = string(machineconfig.X86_64)
+	case "":
+		architectureFileName = string(machineconfig.X86_64) // default to X86_64 if unspecified
+	default:
+		architectureFileName = o.CPUArchitecture
+	}
+	return fmt.Sprintf(isoNameFormat, architectureFileName)
+}
+
```

Comment:
Also this block should be removed, as it requires a knowledge at oc level that it'd be better to avoid. The only safe assumption is that the node-joiner will create an ISO file in case of success. I spent some time experimenting with the `rsync` option, and by using the following code it will be possible to copy the generated ISO with wildcards:

```
	rsyncOptions := &rsync.RsyncOptions{
		Namespace:     o.nodeJoinerNamespace.GetName(),
		Source:        &rsync.PathSpec{PodName: o.nodeJoinerPod.GetName(), Path: "/assets/"},
		ContainerName: nodeJoinerContainer,
		Destination:   &rsync.PathSpec{PodName: "", Path: o.AssetsDir},
		Client:        o.Client,
		Config:        o.Config,
		Compress:      true,
		RshCmd:        fmt.Sprintf("%s --namespace=%s -c %s", o.rsyncRshCmd, o.nodeJoinerNamespace.GetName(), nodeJoinerContainer),
		IOStreams:     o.IOStreams,
		Quiet:         true,
		RsyncInclude:  []string{"*.iso"},
		RsyncExclude:  []string{"*"},
	}
```

Reply:
Thanks for finding a better and more simpler way to fix the problem. I'll update the PR as the rest of the changes aren't needed.